src/test/java/com/linkedin/kafka/clients/utils/tests/KafkaTestUtils.java

```
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2017 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License").  See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.clients.utils.tests;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.apache.commons.io.FileUtils;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.producer.KafkaProducer;
+
+
+public class KafkaTestUtils {
+  private final static AtomicBoolean SHUTDOWN_HOOK_INSTALLED = new AtomicBoolean(false);
+  private final static Thread SHUTDOWN_HOOK;
+  private final static List<File> FILES_TO_CLEAN_UP = Collections.synchronizedList(new ArrayList<>());
+
+  static {
+    SHUTDOWN_HOOK = new Thread(() -> {
+      for (File toCleanUp : FILES_TO_CLEAN_UP) {
+        if (!toCleanUp.exists()) {
+          continue;
+        }
+        try {
+          FileUtils.forceDelete(toCleanUp);
+        } catch (IOException issue) {
+          issue.printStackTrace(System.err);
```

Maybe rethrow the exception to let the uncaught exception handler handle it.